include/aikido/control/PositionCommandExecutor.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef AIKIDO_CONTROL_POSITIONCOMANDEXECUTOR_HPP_
+#define AIKIDO_CONTROL_POSITIONCOMANDEXECUTOR_HPP_
+#include <Eigen/Dense>
+#include <future>
+
+namespace aikido {
+namespace control {
+
+/// Abstract class for executing position commands.
+class PositionCommandExecutor
+{
+public:
+  virtual ~PositionCommandExecutor() = default;
+
+  /// Execute hand to goalPosition
+  /// \param goalPositions Goal positions
+  /// \return future which becomes available when the execution completes.
```

Comment:
Nit: Note that the command succeeds when movement stops, not when the `DegreeOfFreedom`s reach the specified `goalPositions`. Depending upon the implementation, the command may terminate for a variety of reasons, e.g. because of collision.